README.md

```diff
@@ -7,3 +7,60 @@
 This is a continuous-time simulation with the simplest-possible state and non-exponential transitions. The goal is to make a compact, fastest-possible non-exponential continuous-time simulation.
 
 The simplest-possible state is a vector-addition system, which has a state that is a vector of integers just like chemical equations. A transition is defined by a vector of integers that represent how many values to take or give. However, for this simulation, the transition rates can be any function of the state and have exponential or non-exponential distributions.
+
+## Vector Addition System
+
+A [vector addition system](https://en.wikipedia.org/wiki/Vector_addition_system) (VAS) is a simple mathematical language to model systems. It is simpler than the Petri net language, but has the advantage of allowing for more free-form simulation design, as intensity functions can be arbitrary functions of state.
+
+```mermaid
+classDiagram
+
+class VectorAdditionSystem
+    VectorAdditionSystem: +Matrix take
+    VectorAdditionSystem: +Matrix give
+    VectorAdditionSystem: +Vector rates
+
+
+class VectorAdditionModel
+    VectorAdditionModel: +VectorAdditionSystem vas
+    VectorAdditionModel: +Vector state
+    VectorAdditionModel: +Float64 when
+
+class VectorAdditionFSM
+    VectorAdditionFSM: +VectorAdditionModel vam
+    VectorAdditionFSM: +Any sampler
+end
+
+VectorAdditionModel *-- VectorAdditionSystem
+VectorAdditionFSM *-- VectorAdditionModel
+```
+
+Simulation using the VAS requires concrete implementation of the interface:
+
```

Comment:
Yeah, this is not nearly enough to explain what those functions are. It does help me get an overview.